src/palace/manager/api/overdrive/api.py

```diff
@@ -730,7 +742,9 @@ def patron_request(
                 )
             else:
                 # Refresh the token and try again.
-                self.refresh_patron_access_token(patron_credential, patron, pin)
+                self._refresh_patron_oauth_token(
+                    patron_credential, patron, pin, palace_context=palace_context
+                )
```

Comment:
Bug fix here, we weren't propagating the `palace_context` on token refresh, so we would have been using the wrong context in these cases.